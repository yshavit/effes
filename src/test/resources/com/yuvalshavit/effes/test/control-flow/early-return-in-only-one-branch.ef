boolId (bool: True | False) -> True | False:
  case bool of
    True:
      debugPrint True
      return True
    False:
      debugPrint False
  debugPrint Void
  return False

main: debugPrint boolId True
