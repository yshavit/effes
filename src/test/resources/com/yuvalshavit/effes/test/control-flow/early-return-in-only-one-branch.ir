def boolId (bool: False | True) -> False | True:
  CaseStatement
    case:
      VarExpression
        arg: bool
        pos: 0
    of True:
      MethodInvoke
        method: print [built-in]
        arg0:
          CtorInvoke
            type: True
      ReturnStatement
        CtorInvoke
          type: True
    of False:
      MethodInvoke
        method: print [built-in]
        arg0:
          CtorInvoke
            type: False
  MethodInvoke
    method: print [built-in]
    arg0:
      CtorInvoke
        type: Void
  ReturnStatement
    CtorInvoke
      type: False
