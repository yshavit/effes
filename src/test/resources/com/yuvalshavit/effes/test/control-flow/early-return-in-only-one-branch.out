Your argument was: True
Your argument was: True
>> null
